pkg/helpers/tokencmd/basicauth_test.go

```diff
@@ -101,27 +101,17 @@ Password: `,
 		"interactive challenge": {
 			Handler: &BasicChallengeHandler{
 				Host:     "myhost",
-				Reader:   bytes.NewBufferString("myuser\nmypassword\n"),
 				Username: "",
 				Password: "",
 			},
 			Challenges: []Challenge{
-				{
-					Headers:           basicChallenge,
-					ExpectedCanHandle: true,
-					ExpectedHeaders:   http.Header{AUTHORIZATION: []string{getBasicHeader("myuser", "mypassword")}},
-					ExpectedHandled:   true,
-					ExpectedErr:       nil,
-					ExpectedPrompt: `Authentication required for myhost (myrealm)
-Username: Password: `,
-				},
 				{
 					Headers:           basicChallenge,
 					ExpectedCanHandle: true,
 					ExpectedHeaders:   nil,
 					ExpectedHandled:   false,
-					ExpectedErr:       nil,
-					ExpectedPrompt:    ``,
+					ExpectedErr:       BasicNoUsernameError(),
+					ExpectedPrompt:    "",
 				},
 			},
 		},
```

Comment:
this now equals `"non-interactive with no defaults"` and you're testing the only other possible scenario with `"interactive challenge with default user"` so you can remove this test case

Reply:
removed "interactive challenge" case,